### PR TITLE
Add core classes and models for the new (experimental) databases panel

### PR DESCRIPTION
**Pull request by charisk**
This PR introduces some skeleton classes and some core models for the new databases panel. 

It also introduces a `DbModule`, a class that can be used by `extension.ts` to do all the initiatilisation needed. This is a bit of an experiment in itself - trying to keep the extension.ts changes to a minimum.

Most of the code isn't fleshed out yet and I expect changes in both the various classes as well as the models, but I think it's worth getting them in so that we can build on them and hopefully parallelise some of the work.

## Checklist
N/A - internal only:
- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

**Review by norascheuch — APPROVED** [review]
lgtm as far as I can tell, am a but unsure about the db tree data provider (btw is it intentional that in the ui folder `database` is not abbreviated?) but I think this will become clearer with more UI PRs.

**Comment by charisk** [comment]
> btw is it intentional that in the ui folder `database` is not abbreviated?

It was not intentional, thanks for catching!

File: extensions/ql-vscode/src/databases/db-module.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+export class DbModule extends DisposableObject {
+  public async initialize(
+    extensionContext: vscode.ExtensionContext
+  ): Promise<void> {
+    if (extensionContext.extensionMode !== vscode.ExtensionMode.Development ||
+      !isNewQueryRunExperienceEnabled()) {
+      // Currently, we only want to expose the new database panel when we
+      // are in development mode and the developer has enabled the new 
+      // query run experience.
```
Comment (shati-patel) [inline extensions/ql-vscode/src/databases/db-module.ts:17]: Do we also need to check `canary` here? 🤔 
https://github.com/github/vscode-codeql/blob/26e2021551d34a00fab3865147435f06929d910a/extensions/ql-vscode/package.json#L1213-L1215
(That's how we define the panel in the package.json, but I'm not sure how that interacts with the `DbModule` here.)

**Review by shati-patel — APPROVED** [review]
One question, but otherwise LGTM 🐸

File: extensions/ql-vscode/src/databases/db-module.ts (hunk 1 above)
Comment (charisk) [reply extensions/ql-vscode/src/databases/db-module.ts:17]: Good shout, thanks!

File: extensions/ql-vscode/src/databases/db-module.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+export class DbModule extends DisposableObject {
+  public async initialize(
+    extensionContext: vscode.ExtensionContext
+  ): Promise<void> {
+    if (extensionContext.extensionMode !== vscode.ExtensionMode.Development ||
+      !isCanary() ||
+      !isNewQueryRunExperienceEnabled()) {
+      // Currently, we only want to expose the new database panel when we
+      // are in development and canary modem and the developer has enabled the
```
Comment (shati-patel) [inline extensions/ql-vscode/src/databases/db-module.ts:17]: Very minor 😉 
```suggestion
      // are in development and canary mode and the developer has enabled the 
```

**Pull request merged**